packages/salesforcedx-vscode-automation-tests/test/specs/debugApexTests.e2e.ts

```diff
@@ -0,0 +1,242 @@
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {
+  TestReqConfig,
+  ProjectShapeOption,
+  pause,
+  Duration
+} from '@salesforce/salesforcedx-vscode-test-tools/lib/src/core';
+import { log } from '@salesforce/salesforcedx-vscode-test-tools/lib/src/core/miscellaneous';
+import { createApexClassWithTest } from '@salesforce/salesforcedx-vscode-test-tools/lib/src/salesforce-components';
+import {
+  continueDebugging,
+  getTestsSection,
+  verifyTestItemsInSideBar
+} from '@salesforce/salesforcedx-vscode-test-tools/lib/src/testing';
+import { TestSetup } from '@salesforce/salesforcedx-vscode-test-tools/lib/src/testSetup';
+import {
+  executeQuickPick,
+  notificationIsPresentWithTimeout,
+  getWorkbench,
+  getStatusBarItemWhichIncludes,
+  getTextEditor,
+  dismissAllNotifications
+} from '@salesforce/salesforcedx-vscode-test-tools/lib/src/ui-interaction';
+import { expect } from 'chai';
+import { step } from 'mocha-steps';
+import { TreeItem, after } from 'vscode-extension-tester';
+
+describe('Debug Apex Tests', async () => {
+  let testSetup: TestSetup;
+  const testReqConfig: TestReqConfig = {
+    projectConfig: {
+      projectShape: ProjectShapeOption.NEW
+    },
+    isOrgRequired: true,
+    testSuiteSuffixName: 'DebugApexTests'
+  };
+
+  step('Set up the testing environment', async () => {
+    log('DebugApexTests - Set up the testing environment');
+    testSetup = await TestSetup.setUp(testReqConfig);
+
+    // Create Apex class 1 and test
+    try {
+      await createApexClassWithTest('ExampleApexClass1');
+    } catch (error) {
+      await createApexClassWithTest('ExampleApexClass1');
+    }
+
+    // Create Apex class 2 and test
+    try {
+      await createApexClassWithTest('ExampleApexClass2');
+    } catch (error) {
+      await createApexClassWithTest('ExampleApexClass2');
+    }
+
+    // Push source to org
+    await executeQuickPick('SFDX: Push Source to Default Org and Ignore Conflicts', Duration.seconds(1));
+
+    // Look for the success notification that appears which says, "SFDX: Push Source to Default Org and Ignore Conflicts successfully ran".
+    let successPushNotificationWasFound;
+    try {
+      successPushNotificationWasFound = await notificationIsPresentWithTimeout(
+        /SFDX: Push Source to Default Org and Ignore Conflicts successfully ran/,
+        Duration.TEN_MINUTES
+      );
+      expect(successPushNotificationWasFound).to.equal(true);
+    } catch (error) {
+      await getWorkbench().openNotificationsCenter();
+      successPushNotificationWasFound = await notificationIsPresentWithTimeout(
+        /SFDX: Push Source to Default Org and Ignore Conflicts successfully ran/,
+        Duration.TEN_MINUTES
+      );
+      expect(successPushNotificationWasFound).to.equal(true);
+    }
+  });
+
+  step('Verify LSP finished indexing', async () => {
```

Comment:
how attached are we to mocha-steps instead of just using `--bail`? or conditionals like https://stackoverflow.com/questions/69188795/mocha-bail-on-one-specific-describe-only

Reply:
eslint seems to really not like this (no-misused-promises)